src/Parser/Shaman/Restoration/Modules/Abilities.js

```diff
@@ -9,21 +10,21 @@ class Abilities extends CoreAbilities {
       {
         spell: SPELLS.RIPTIDE,
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
-        charges: combatant.hasTalent(SPELLS.ECHO_OF_THE_ELEMENTS_TALENT.id) ? 2 : 1,
+        charges: ((combatant.hasTalent(SPELLS.ECHO_OF_THE_ELEMENTS_TALENT.id) || this.selectedCombatant.hasFinger(ITEMS.SOUL_OF_THE_FARSEER.id)) || this.selectedCombatant.hasFinger(ITEMS.SOUL_OF_THE_FARSEER.id)) ? 2 : 1,
```

Comment:
Do you need both Soul of the Farseer checks here? It seems redundant.

Reply:
search & replace fail, thanks